### ISSUE TITLE
AttributeError: module 'torch' has no attribute '_assert'

### ISSUE THREAD

**Issue by sejongpyo** [pyg-team/pytorch_geometric#3710]
## 🐛 Bug

Hi, i keep having a problem even if i followed this issue  https://github.com/pyg-team/pytorch_geometric/issues/3231#issue-1006736099.
```
 File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch_geometric/nn/__init__.py", line 5, in <module>
    from .to_hetero_transformer import to_hetero
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch_geometric/nn/to_hetero_transformer.py", line 11, in <module>
    from torch_geometric.nn.fx import Transformer
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch_geometric/nn/fx.py", line 11, in <module>
    from torch.fx import GraphModule, Graph, Node
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/fx/__init__.py", line 84, in <module>
    from .graph_module import GraphModule
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/fx/graph_module.py", line 8, in <module>
    from .graph import Graph, _is_from_torch, _custom_builtins, PythonCode
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/fx/graph.py", line 1, in <module>
    from .node import Node, Argument, Target, map_arg, _type_repr, _get_qualified_name
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/fx/node.py", line 26, in <module>
    _side_effectful_functions: Set[Callable] = {torch._assert}
AttributeError: module 'torch' has no attribute '_assert'
```
I have reinstalled pytorch and torch-geometric from the master.
This is the AttributeError when i try to import torch.fx
```
>>> import torch
>>> torch.__version__
'1.7.1'
>>> import torch.fx
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/fx/__init__.py", line 84, in <module>
    from .graph_module import GraphModule
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/fx/graph_module.py", line 8, in <module>
    from .graph import Graph, _is_from_torch, _custom_builtins, PythonCode
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/fx/graph.py", line 1, in <module>
    from .node import Node, Argument, Target, map_arg, _type_repr, _get_qualified_name
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/fx/node.py", line 26, in <module>
    _side_effectful_functions: Set[Callable] = {torch._assert}
AttributeError: module 'torch' has no attribute '_assert'
```

### Environment

* PyG version : 2.0.2
* PyTorch version: 1.7.1 (installed by conda install pytorch==1.7.1 torchvision==0.8.2 torchaudio==0.7.2 cudatoolkit=10.1 -c pytorch)
* OS :ubuntu 16.04
* Python version : 3.6
* CUDA/cuDNN version: 10.1
* How you installed PyTorch and PyG : pip
* torch-scatter: 2.0.7
* torch-sparse: 0.6.9

**Comment by rusty1s** [comment]
I added the `AttributeError` to the catch blocks, see https://github.com/pyg-team/pytorch_geometric/commit/c304a283f1c3282c5476c34b77175c0ef076cf0c. Hopefully, this resolves your issues.

**Comment by sejongpyo** [comment]
> I added the `AttributeError` to the catch blocks, see [c304a28](https://github.com/pyg-team/pytorch_geometric/commit/c304a283f1c3282c5476c34b77175c0ef076cf0c). Hopefully, this resolves your issues.

Thank you! The problem is resolved!

**Comment by larosarah** [comment]
Hello, i have the same problem, this is the AttributeError when i try to import torch.fx, can you help me please ? 

![2](https://user-images.githubusercontent.com/109140428/178453864-a125a8bb-065b-4cc7-9567-6b35ebee6c2c.png)
![1](https://user-images.githubusercontent.com/109140428/178453872-4220c7ef-f0f3-4670-87a7-f6e3a6bfdff5.png)

Environment

    PyTorch version: 1.7.1 (installed by conda install pytorch torchvision cudatoolkit=11.7 -c pytorch -c conda-forge)
    Pytorch-lightning: 1.5.10
    OS :ubuntu 16.04
    Python version : 3.9.13
    CUDA/cuDNN version: 11.7
    How you installed : conda
    torch: 1.9.1+cu102
    torchgeo: 0.1.1
    torchmetrics: 0.9.2
    torchvision: 0.10.1+cu102

**Comment by rusty1s** [comment]
Yes, `torch.fx` is not included in PyTorch 1.7.1 so this error is to be expected.

**Comment by itxmarcos** [comment]
I have a similar problem (pytorch 1.7.1, CUDA 10.1, ubuntu 20.04.6 LTS, conda 4.7.11, python 3.9.17)

from groundingdino.util.inference import Model

grounding_dino_model = Model(model_config_path=GROUNDING_DINO_CONFIG_PATH, model_checkpoint_path=GROUNDING_DINO_CHECKPOINT_PATH)

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[12], line 3
      1 get_ipython().run_line_magic('cd', '{HOME}/GroundingDINO')
----> 3 from groundingdino.util.inference import Model
      5 grounding_dino_model = Model(model_config_path=GROUNDING_DINO_CONFIG_PATH, model_checkpoint_path=GROUNDING_DINO_CHECKPOINT_PATH)

File ~/GroundingDINO/groundingdino/util/inference.py:11
      8 from torchvision.ops import box_convert
     10 import groundingdino.datasets.transforms as T
---> 11 from groundingdino.models import build_model
     12 from groundingdino.util.misc import clean_state_dict
     13 from groundingdino.util.slconfig import SLConfig

File ~/GroundingDINO/groundingdino/models/__init__.py:8
      1 # ------------------------------------------------------------------------
      2 # Grounding DINO
      3 # url: https://github.com/IDEA-Research/GroundingDINO
   (...)
      6 # ------------------------------------------------------------------------
      7 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
----> 8 from .GroundingDINO import build_groundingdino
     11 def build_model(args):
     12     # we use register to maintain models from catdet6 on.
     13     from .registry import MODULE_BUILD_FUNCS

File ~/GroundingDINO/groundingdino/models/GroundingDINO/__init__.py:15
      1 # ------------------------------------------------------------------------
      2 # Grounding DINO
      3 # url: https://github.com/IDEA-Research/GroundingDINO
   (...)
     12 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
     13 # ------------------------------------------------------------------------
---> 15 from .groundingdino import build_groundingdino

File ~/GroundingDINO/groundingdino/models/GroundingDINO/groundingdino.py:41
     38 from groundingdino.util.vl_utils import create_positive_map_from_span
     40 from ..registry import MODULE_BUILD_FUNCS
---> 41 from .backbone import build_backbone
     42 from .bertwarper import (
     43     BertModelWarper,
     44     generate_masks_with_special_tokens,
     45     generate_masks_with_special_tokens_and_transfer_map,
     46 )
     47 from .transformer import build_transformer

File ~/GroundingDINO/groundingdino/models/GroundingDINO/backbone/__init__.py:1
----> 1 from .backbone import build_backbone

File ~/GroundingDINO/groundingdino/models/GroundingDINO/backbone/backbone.py:30
     27 from groundingdino.util.misc import NestedTensor, clean_state_dict, is_main_process
     29 from .position_encoding import build_position_encoding
---> 30 from .swin_transformer import build_swin_transformer
     33 class FrozenBatchNorm2d(torch.nn.Module):
     34     """
     35     BatchNorm2d where the batch statistics and the affine parameters are fixed.
     36 
   (...)
     39     produce nans.
     40     """

File ~/GroundingDINO/groundingdino/models/GroundingDINO/backbone/swin_transformer.py:19
     17 import torch.nn.functional as F
     18 import torch.utils.checkpoint as checkpoint
---> 19 from timm.models.layers import DropPath, to_2tuple, trunc_normal_
     21 from groundingdino.util.misc import NestedTensor
     24 class Mlp(nn.Module):

File /data/anaconda5/envs/PAINTEC/lib/python3.8/site-packages/timm/__init__.py:3
      1 from .version import __version__
      2 from .layers import is_scriptable, is_exportable, set_scriptable, set_exportable
----> 3 from .models import create_model, list_models, list_pretrained, is_model, list_modules, model_entrypoint, \
      4     is_model_pretrained, get_pretrained_cfg, get_pretrained_cfg_value

File /data/anaconda5/envs/PAINTEC/lib/python3.8/site-packages/timm/models/__init__.py:1
----> 1 from .beit import *
      2 from .byoanet import *
      3 from .byobnet import *

File /data/anaconda5/envs/PAINTEC/lib/python3.8/site-packages/timm/models/beit.py:52
     49 from timm.data import IMAGENET_DEFAULT_MEAN, IMAGENET_DEFAULT_STD
     50 from timm.layers import PatchEmbed, Mlp, SwiGLU, LayerNorm, DropPath, trunc_normal_, use_fused_attn
---> 52 from ._builder import build_model_with_cfg
     53 from ._registry import generate_default_cfgs, register_model
     54 from .vision_transformer import checkpoint_filter_fn

File /data/anaconda5/envs/PAINTEC/lib/python3.8/site-packages/timm/models/_builder.py:11
      8 from torch.hub import load_state_dict_from_url
     10 from timm.models._features import FeatureListNet, FeatureHookNet
---> 11 from timm.models._features_fx import FeatureGraphNet
     12 from timm.models._helpers import load_state_dict
     13 from timm.models._hub import has_hf_hub, download_cached_file, check_cached_file, load_state_dict_from_hf

File /data/anaconda5/envs/PAINTEC/lib/python3.8/site-packages/timm/models/_features_fx.py:12
      9 from ._features import _get_feature_info
     11 try:
---> 12     from torchvision.models.feature_extraction import create_feature_extractor as _create_feature_extractor
     13     has_fx_feature_extraction = True
     14 except ImportError:

File /data/anaconda5/envs/PAINTEC/lib/python3.8/site-packages/torchvision/models/feature_extraction.py:10
      8 import torch
      9 from torch import nn
---> 10 from torch import fx
     11 from torch.fx.graph_module import _copy_attr
     14 __all__ = ['create_feature_extractor', 'get_graph_node_names']

File /data/anaconda5/envs/PAINTEC/lib/python3.8/site-packages/torch/fx/__init__.py:83
      1 r'''
      2 FX is a toolkit for developers to use to transform ``nn.Module``
      3 instances. FX consists of three main components: a **symbolic tracer,**
   (...)
     80 repository.
     81 '''
---> 83 from .graph_module import GraphModule
     84 from ._symbolic_trace import symbolic_trace, Tracer, wrap, PH, ProxyableClassMeta
     85 from .graph import Graph

File /data/anaconda5/envs/PAINTEC/lib/python3.8/site-packages/torch/fx/graph_module.py:8
      6 import linecache
      7 from typing import Type, Dict, List, Any, Union, Optional, Set
----> 8 from .graph import Graph, _is_from_torch, _custom_builtins, PythonCode
      9 from ._compatibility import compatibility
     10 from torch.package import Importer, sys_importer

File /data/anaconda5/envs/PAINTEC/lib/python3.8/site-packages/torch/fx/graph.py:1
----> 1 from .node import Node, Argument, Target, map_arg, _type_repr, _get_qualified_name
      2 import torch.utils._pytree as pytree
      3 from . import _pytree as fx_pytree

File /data/anaconda5/envs/PAINTEC/lib/python3.8/site-packages/torch/fx/node.py:28
     16 Target = Union[Callable[..., Any], str]
     18 Argument = Optional[Union[
     19     Tuple[Any, ...],  # actually Argument, but mypy can't represent recursive types
     20     List[Any],  # actually Argument
   (...)
     24     BaseArgumentTypes
     25 ]]
     27 _side_effectful_functions: Set[Callable] = {
---> 28     torch._assert, torch.ops.profiler._record_function_enter,
     29     torch.ops.profiler._record_function_exit}
     31 # this is fixed on master, WAR for 1.5
     32 def _find_module_of_method(orig_method: Callable[..., Any]) -> str:

AttributeError: module 'torch' has no attribute '_assert'

**Comment by akihironitta** [comment]
@sejongpyo As far as I looked at your stack trace, the error seems completely unrelated to PyG, but it looks like you're using too old PyTorch. `torch._assert` was added in PyTorch 1.10, so I'd suggest updating your PyTorch. https://pytorch.org/docs/1.10/generated/torch._assert.html
